internal/admin/admin_test.go

```diff
@@ -223,6 +226,335 @@ func TestAdmin_GetStorageNode(t *testing.T) {
 	}
 }
 
+// This test verifies whether the expected logstreams are sealed
+// in the event of an RPC timeout due to a storagenode failure.
+// In addition, it verifies that the logstreams on the healthy node
+// is not sealed in the event of intermittent failure of the mr request in that situation.
+func TestAdmin_Storagenode_HeartbeatTimeout_Seal(t *testing.T) {
+	const (
+		cid         = types.ClusterID(1)
+		tpid        = types.TopicID(1)
+		victimsnid  = types.StorageNodeID(1)
+		normalsnid  = types.StorageNodeID(2)
+		victimlsid1 = types.LogStreamID(1)
+		victimlsid2 = types.LogStreamID(2)
+		normallsid1 = types.LogStreamID(3)
+		normallsid2 = types.LogStreamID(4)
+		//TODO:: it should be 2
+		expectedSealed = 1
+		// rpcTimeout should be greater than snwatcher hb timeout(1s)
+		rpcTimeout = 2 * time.Second
+		// rate to cause intermittent failures
+		mrReqFailRate = 30
+	)
+
+	type testEnv struct {
+		mrfail      atomic.Bool
+		failedsnid  atomic.Int32
+		sealcnt     atomic.Int32
+		sealedlsids sync.Map
+		normallsids set.Set
+		victimlsids set.Set
+	}
+
+	tcs := []struct {
+		name  string
+		testf func(t *testing.T, env *testEnv)
+	}{
+		{
+			name: "ExpectedSeal",
+			testf: func(t *testing.T, env *testEnv) {
+				// Make victimsn failed
+				env.failedsnid.Store(int32(victimsnid))
+
+				require.Eventually(t, func() bool {
+					return env.sealcnt.Load() >= int32(env.victimlsids.Size()+env.normallsids.Size())
+				}, 30*time.Second, time.Second)
+
+				var numSealed int
+				env.sealedlsids.Range(func(k, _ any) bool {
+					numSealed++
+					require.True(t, env.victimlsids.Contains(k.(types.LogStreamID)))
+					return true
+				})
+
+				require.Equal(t, expectedSealed, numSealed)
```

Comment:
In my understanding, this commit provides a new test that manifests a bug, right?
In this case, `env.sealedlsids` should have `victimlsid1` and `victimlsid2`. If I am on the wrong way, please give me feedback.

Reply:
That's right. This is a test to check if there is a bug. After fixing the bug, we will change the expected value and convert it to the desired test. Then `env.sealedlsids` should contain `victimlsid1` and `victimlsid2`.